### ISSUE TITLE
iOS App will not compile - UriBeacon missing

### ISSUE THREAD

**Issue by gemtot** [google/physical-web#483]
App imports UriBeacon/UriBeacon.h but UriBeacon is not included in the repo.

**Comment by kartikgaade** [comment]
Hi,

I have the same issue. Can you please let me know how to resolve the "UriBeacon/UriBeacon.h file not found" issue

**Comment by cco3** [comment]
Have you pulled the submodules?

**Comment by kartikgaade** [comment]
Yes, No luck yet! Can you please provide the steps... I might be missing something here

**Comment by dinhviethoa** [comment]
Use the following:
```
$ git submodule init
$ git submodule update
```

**Comment by mmocny** [comment]
Pro tip:
`git submodule update --init --recursive`

**Comment by kartikgaade** [comment]
I pulled all the submodules but the project does not build for some reason. If someone was able to successfully build/run the physical web project, can you please share the download link? I have gone crazy resolving this issue

**Comment by dinhviethoa** [comment]
I just fixed the UriBeacon dependency. Could you try again? Thanks.

**Comment by kartikgaade** [comment]
I am getting different errors now.
<img width="477" alt="screen shot 2015-09-16 at 2 11 32 pm" src="https://cloud.githubusercontent.com/assets/4049911/9913822/c4a1c32a-5c7c-11e5-858d-0c20a7c50667.png">

**Comment by dinhviethoa** [comment]
Could you paste in a gist a complete log of the build? It will be helpful to analyze your issue.
Possible issue:
- the xcodeproject has been modified and you need to revert it
- there's a broken version of the build of UriBeacon library around and you need to clean it

**Comment by kartikgaade** [comment]
<img width="1395" alt="screen shot 2015-09-16 at 2 40 27 pm" src="https://cloud.githubusercontent.com/assets/4049911/9914642/c7967f90-5c80-11e5-810d-38929f106a83.png">

**Comment by jeffprestes** [comment]
Kartik,

meanwhile Hoa fix this, I've created a workaround in my fork. You can see
it at : https://github.com/jeffprestes/physical-web/tree/master/ios

Hoa,

I believe the iOS project structure is tied to the machine of who has
permission and the keys to submit it to AppStore. However, I and other devs
just want to run it from our machine for tests and learning purposes. Is
there a way to change project structure to make it more "machine
independent"?

Thanks,

-Jeff




2015-09-16 15:39 GMT-03:00 kartikgaade <notifications@github.com>:

**Comment by dinhviethoa** [comment]
@jeffprestes I'm not sure. Could you show me which part of the project is dependent on my machine?
I tried to check in code that are independent of my dev environment.
Could you highlight specific issues? It will help me fix those if that's the case.
Thanks.

**Comment by jeffprestes** [comment]
Hi Hoa,

Sorry for the late response. I got married last month and I was in
Honeymoon.
So, I've tried again and now I succeed. Looks like your last upgrades fixed
the problem. Thanks!
However, it worked using this command after the clone: "git pull --rebase
--prune && git submodule update --init --recursive" it was suggested by
Michal. I know it maybe sound obvious but for who is not familiar with the
project can struggle to fix the dependencies. I'd suggest you to include
this line at README.md

If you want, I've updated it in my fork and I can submit a pull request

Again thanks for the updates.

-Jeff

2015-09-17 12:00 GMT-07:00 Hoà V. DINH <notifications@github.com>:
